### ISSUE TITLE
Vendor:publish doesn't create migration

### ISSUE THREAD

**Issue by dambridge** [mpociot/teamwork#59]
Running php artisan vendor:publish --provider="Mpociot\Teamwork\TeamworkServiceProvider" publishes the config file but not the migration. Am on 5.3.18

**Comment by etherealite** [comment]
+1

It registers its own migration folder that always runs and changes it's table and field names according to the `config/teamwork.php`. I'm not a fan of configurable migrations personally. This approach didn't work for me at all since I don't want to use the provided migrations at all, and substitute them with my own.

What I ended up doing was overriding the service provider in my `app/providers`

``` php
<?php

namespace App\Providers;

use Illuminate\Support\ServiceProvider;
use Mpociot\Teamwork\TeamworkServiceProvider;

class WingzTeamworkServiceProvider extends TeamworkServiceProvider
{
    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        $this->publishConfig();
    }

}
```

Slightly annoying, Id prefer that the migrations were simply published rather than forcing me to run them. Going further, the `teamwork.php` config file allows you to change the name of the tables, but not the name of the foreign keys for the users and invites tables. Why would I want to change the name of the teams table, yet leave the references to its id named 'team_id' on the tables that refer back to it? What then is the point of having the table name be configurable?

**Comment by dayio** [comment]
+1
I aggree with both of you, i need to use this plugin but i want to have a control over migrations.

It's important to have things clear on this side, because if i need to add a field or change the Model for another in the config file, the migration file will be run instead of using one of mine.

**Comment by mpociot** [comment]
Fixed in 4.0.2
